### PR TITLE
[ISSUE-1126] update dependencies

### LINKED ISSUE #1126: Upgrade outdated dependencies
**Is your feature request related to a problem? Please describe.**
There are multiple outdated dependencies, some even two years old.

**Describe the solution you'd like**
Outdated dependencies updated.

**Describe alternatives you've considered**
N/A

**Additional context**
N/A

### PR DESCRIPTION
**Pull request by rafalw82**
### Resolves #1126 

Updating golang dependencies, and the code to adjust new api requirements.

## PR checklist
- [ ] Add link to the issue
- [ ] Choose Project
- [ ] Choose PR label
- [ ] New unit tests added
- [ ] Modified code has meaningful comments
- [ ] All TODOs are linked with the issues
- [ ] All comments are resolved

## Testing
All regression

### CODE REVIEW

File: pkg/base/k8s/kubecache_test.go [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+package k8s
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestKubeCache_InitKubeCache(t *testing.T) {
+	// testLogger = logrus.New()
```
Comment (Kacper-Michon-DELL) [inline pkg/base/k8s/kubecache_test.go:26]: Is this commented code relevant? If not, please remove

File: pkg/base/k8s/kubecache_test.go [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+package k8s
+
+import (
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+)
+
+func TestKubeCache_InitKubeCache(t *testing.T) {
```
Comment (Andrzej-Zukowski) [inline pkg/base/k8s/kubecache_test.go:25]: This test does not test anything

File: pkg/crcontrollers/node/controller_test.go [hunk 3]
```diff
@@ -371,11 +394,17 @@ func Test_reconcileForCSIBMNode(t *testing.T) {
...
+		bmNode.Finalizers = []string{"csibm-finalizer"}
 
 		createObjects(t, c.k8sClient, bmNode, k8sNode)
 
-		res, err := c.reconcileForCSIBMNode(bmNode)
+		err := c.k8sClient.DeleteCR(testCtx, bmNode)
+		assert.Nil(t, err)
+
+		delBmNode := new(nodecrd.Node)
+		c.k8sClient.Get(testCtx, k8sCl.ObjectKey{Name: bmNode.Name}, delBmNode)
```
Comment (Andrzej-Zukowski) [inline pkg/crcontrollers/node/controller_test.go:405]: Shouldn't we test also err ?

File: pkg/crcontrollers/node/controller_test.go [hunk 4]
```diff
@@ -470,11 +499,17 @@ func Test_reconcileForCSIBMNode(t *testing.T) {
...
 		bmNode.Finalizers = []string{"csibm-finalizer"}
 
 		createObjects(t, c.k8sClient, bmNode)
-		res, err := c.reconcileForCSIBMNode(bmNode)
+
+		err := c.k8sClient.DeleteCR(testCtx, bmNode)
+		assert.Nil(t, err)
+
+		delBmNode := new(nodecrd.Node)
+		c.k8sClient.Get(testCtx, k8sCl.ObjectKey{Name: bmNode.Name}, delBmNode)
```
Comment (Andrzej-Zukowski) [inline pkg/crcontrollers/node/controller_test.go:510]: Same as above

File: pkg/base/k8s/kubecache_test.go (hunk 1 above)
Comment (rafalw82) [reply pkg/base/k8s/kubecache_test.go]: fixed

File: pkg/base/k8s/kubecache_test.go (hunk 2 above)
Comment (rafalw82) [reply pkg/base/k8s/kubecache_test.go]: fixed

**Review by Kacper-Michon-DELL — APPROVED** [review]

**Review by Andrzej-Zukowski — APPROVED** [review]

**Pull request merged**